official/recommendation/ncf_keras_main.py

```diff
@@ -355,8 +359,12 @@ def step_fn(features):
             softmax_logits,
             sample_weight=features[rconst.VALID_POINT_MASK])
         loss *= (1.0 / params["batch_size"])
+        if FLAGS.dtype == "fp16":
```

Comment:
does this support fp16 only in the CTL path? 
or does wrapping the optimizer change this for the compile/fit version too?

Reply:
This should support both (havent tested comp/fit yet). As compile/fit does not do any special gradients calculation we dont need to do any manual loss scaling that we are doing in the CTL version which uses gradient.tape()

Reply:
just tested compile/fit and it work with this API as well.

Reply:
hm, i still don't understand why CTL needs the loss scaling but in compile fit we don't ?

Reply:
https://www.tensorflow.org/api_docs/python/tf/train/experimental/enable_mixed_precision_graph_rewrite
If we use "opt.minimize() or opt.compute_gradients()" to compute gradients which compile/fit does internally, then the optimizer does the loss scaling for us. But if we use Tf.GradientTape like in CTL to compute gradients, then we need to do the scaling manually.